code/datums/sleep_adv/sleep_adv.dm

```diff
@@ -96,7 +96,8 @@
 		trait_capped_level = SKILL_LEVEL_LEGENDARY
 	#endif
 
-	if(trait_capped_level && (trait_capped_level <= mind.current.get_skill_level(skill)))
+	// Using this prevent a bug where you can bank xp to go one beyond cap
+	if(trait_capped_level && enough_sleep_xp_to_advance(skill, trait_capped_level - mind.current.get_skill_level(skill)))
```

Comment:
its capped to 4
my current skill is 3

how will this stop me from banking two levels to 5?